### PR TITLE
Add Workflow Matrix

### PR DESCRIPTION
**Pull request by gagelarsen**
Changes Made:
- Added Matrix to Build Workflow to test on versions of python and django
- Updated install script to have args for versions of python and django

### CODE REVIEW

**Review by shawncrawley — APPROVED** [review]
Looks as good as anything can to a blind person... JK, sorta. I have no tangible experience with these configs - but they look properly/syntactically correct and my assumptions about how they work don't reveal any glaring issues.

File: .github/workflows/tethys-release.yml [hunk 1]
```diff
@@ -76,13 +77,14 @@ jobs:
...
-    name: Conda Build (${{ matrix.platform }})
+    name: Conda Build (${{ matrix.platform }}, ${{ matrix.django-version }}, ${{ matrix.python-version }})
     runs-on: ${{ matrix.platform }}
     strategy:
       fail-fast: false
       matrix:
         platform: [ubuntu-latest]
-        python-version: [3.10]
+        python-version: ["3.10", "3.11", "3.12"]
+        django-version: ["3.2", "4.2", "5.0"]
```
Comment (swainn) [inline .github/workflows/tethys-release.yml]: I don't think that the matrix is necessary with the conda build. We probably just want to generate a package with python and django unpinned.

File: scripts/install_tethys.sh [hunk 2]
```diff
@@ -88,6 +90,8 @@ TETHYS_DB_SUPER_PASSWORD='pass'
 TETHYS_DB_PORT=5436
 TETHYS_DB_DIR='psql'
 CONDA_HOME=~/miniconda
+PYTHON_VERSION='3.12'
+DJANGO_VERSION='4.0'
```
Comment (swainn) [inline scripts/install_tethys.sh]: Change default to latest LTS: 4.2

Comment (gagelarsen) [reply scripts/install_tethys.sh]: done

File: .github/workflows/tethys-release.yml (hunk 1 above)
Comment (gagelarsen) [reply .github/workflows/tethys-release.yml]: done

**Comment by swainn** [comment]
@sdc50 we need to decided what gets tagged as "latest" and "dev" for the Docker build. I think the way it works out now, it is whatever finishes last. Also what does matrix build mean for the conda build?

**Review by sdc50 — APPROVED** [review]

File: scripts/install_tethys.sh [hunk 3]
```diff
@@ -351,6 +363,18 @@ then
         git checkout ${BRANCH}
     fi
 
+    if [ -n "${DJANGO_VERSION}" ]
+    then
+        echo "Updating environment.yml Django version ${DJANGO_VERSION}..."
+        sudo sed -i.bak "s/django>=.*/django>=${DJANGO_VERSION}/" "${TETHYS_SRC}/environment.yml"
```
Comment (swainn) [inline scripts/install_tethys.sh:369]: We also need to update the micro_environment.yml.

**Review by swainn — COMMENTED** [review]
Update what is published as follows:

# Production (tethys-release.yml)
- Docker: publish each version in the build matrix with tag `<python>-<django>-<tethys>`
- Docker: don't publish/build/tag a latest version
- Conda: publish one version, tagged with current release version, without pins on Django or Python (other than the lower bound on Python).

# Dev (tethys.yml)
- Docker: publish the dev images with tagging scheme `<python>-<djano>-dev`
- Conda: revert back to single build with no extra pins for Python or Django and publish to the tethysplatform/label/dev channel on anaconda.

File: .github/workflows/tethys-release.yml [hunk 4]
```diff
@@ -119,7 +114,7 @@ jobs:
       - name: Setup Tethys
         run: |
           cd ..
-          bash ./tethys/scripts/install_tethys.sh --partial-tethys-install me -n tethys -s $PWD/tethys -x
+          bash ./tethys/scripts/install_tethys.sh --partial-tethys-install me -n tethys -s $PWD/tethys -x  -p "3.8"
```
[the remark this answers is not in the document]
Comment (gagelarsen) [reply .github/workflows/tethys-release.yml]: removed the pin

File: .github/workflows/tethys-release.yml [hunk 5]
```diff
@@ -12,13 +12,14 @@ env:
...
-    name: Docker Build (${{ matrix.platform }})
+    name: Docker Build (${{ matrix.platform }}, ${{ matrix.django-version }}, ${{ matrix.python-version }})
     runs-on: ${{ matrix.platform }}
     strategy:
       fail-fast: false
       matrix:
         platform: [ubuntu-latest]
-        python-version: [3.10]
+        python-version: ["3.10", "3.11", "3.12"]
+        django-version: ["3.2", "4.2", "5.0"]
```
[the remark this answers is not in the document]
Comment (swainn) [reply .github/workflows/tethys-release.yml:22]: Don't worry about it then.

File: .github/workflows/tethys.yml [hunk 6]
```diff
@@ -85,7 +85,7 @@ jobs:
           tethys test -c -u -v 2
       # Generate Coverage Report
       - name: Generate Coverage Report
-        if: matrix.platform == 'ubuntu-latest'
+        if: ${{ matrix.platform == 'ubuntu-latest' && matrix.python-version == '3.10' && matrix.django-version == '4.2' }}
```
[the remark this answers is not in the document]
Comment (swainn) [reply .github/workflows/tethys.yml:88]: Let's keep it how it is.

**Pull request merged**